app/services/payment_providers/stripe/refresh_webhook_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module PaymentProviders
+  module Stripe
+    class RefreshWebhookService < BaseService
+      Result = BaseResult
+
+      def call
+        ::Stripe::WebhookEndpoint.update(
+          payment_provider.webhook_id,
+          webhook_endpoint_shared_params,
+          {api_key:}
+        )
+
+        result
+      rescue ActiveRecord::RecordInvalid => e
+        result.record_validation_failure!(record: e.record)
```

Comment:
Maybe I am missing something, but it seems that we are not touching any model. So maybe these two lines are not required.

Reply:
Indeed! I'll remove in the next PR (toi avoid conflict and stuff) Thanks!